### DISCUSSION TITLE
APIs of Telemetry Service

### DISCUSSION THREAD

**Discussion by kameshbhr** [Sunbird-Obsrv/Community#2] [Query]
Hi, 
I would like to know where I can get the API documentation of the Telemetry Service.
Thanks.

**Comment by pramodkvarma** [comment]
Have you checked https://obsrv.sunbird.org pages? It was supposed to be linked from there (both telemetry specs and the obsrv APIs). @rhwarrier can you see if obsrv APIs are documented and if so, provide link? If not, we need that completed.

**Reply by kameshbhr** [reply]
I checked in obsrv sunbird pages. I could see the link to the telemetry specs but not to telemetry service APIs. I can see links to other APIs (like data service).

**Reply by pramodkvarma** [reply]
Yes even I noticed.

@rhwarrier @surendrasinghs can either of you take it up please?

**Reply by rhwarrier** [reply]
Looks like the documentation for the telemetry service API needs to get done - there is only one API, and it needs to get documented as a swagger doc. We'll get it done and get back on this thread. (and also update the Obsrv microsite)

**Reply by pramodkvarma** [reply]
Ok. As I understand there is one server side API to push telemetry, then there are APIs to process, and then there are client SDK APIs all part of obsrv software building block. We must ensure developers who want to adopt Obsrv Building Block indeed can learn all aspects. Please get the engineers working on this to get intial docs up and running. Thanks.

**Reply by anandp504** [reply]
@pramodkvarma @kameshbhr 
Please find the swagger documentation for all the APIs as part of Sunbird Obsrv below:
1. [Telemetry Service API](http://docs.sunbird.org/latest/apis/telemetryapi/index.html)
2. [Report APIs](http://docs.sunbird.org/latest/apis/reports/)
3. [Data Exhaust API](http://docs.sunbird.org/latest/apis/dataexhaustapi/index.html)
4. [Druid Report API](http://docs.sunbird.org/latest/apis/druidreportapi/index.html)

The Sunbird portal and the mobile app use different methodology to generate telemetry. The Sunbird portal uses the nodejs package [Telemetry SDK](https://www.npmjs.com/package/@project-sunbird/telemetry-sdk) and is currently being mainted by portal team. This probably needs to go under the Telemetry Building block. The mobile app uses its own framework to generate the events.

We will have all the relevant documentation for the Sunbird Obsrv Building Block to be put in one place so that it will ease out the adoption. Thanks.

**Reply by pramodkvarma** [reply]
Fantastic! I have seen them earlier. 

Btw those URLs didn't work for me when I clicked them.  I am getting error saying:

`This site can’t be reached 
docs.sunbird.org refused to connect.`

Please move these docs to appropriate building block docs. 

Thanks Anand.
